src/systems/joint_state_publisher/JointStatePublisher.cc

```diff
@@ -142,11 +151,19 @@ void JointStatePublisher::PostUpdate(const UpdateInfo &_info,
       worldName = _ecm.Component<components::Name>(
           parentEntity->Data())->Data();
 
-      // Advertise the state topic
-      std::string topic = std::string("/world/") + worldName + "/model/"
-        + this->model.Name(_ecm) + "/joint_state";
+      // if topic not set it will be empty
+      std::vector<std::string> topics;
+      // this helps avoid unecesarry invalid topic error
+      if (!this->topic.empty())
+      {
+        topics.push_back(this->topic);
+      }
+      topics.push_back(std::string("/world/") + worldName + "/model/"
+        + this->model.Name(_ecm) + "/joint_state");
+      this->topic = validTopic(topics);
```

Comment:
you should check if the topic is empty

```suggestion
      this->topic = validTopic(topics);
      if (this->topic.empty())
      {
      show and error and maybe return ?
      }
```

Reply:
Hi, correct me if I am wrong but I am not sure I understand why this check might be required. 
There are two possible conditions: 
 * The user provides a topic in the sdf
 * The user doesn't provide a topic and goes with default one

In the first one, `this->topic` will be initially set to the value entered by user and the `topics` vector will have [2 elements](https://github.com/jasmeet0915/ign-gazebo/blob/5105afb2f4275987239c2455e964db4e336b46dd/src/systems/joint_state_publisher/JointStatePublisher.cc#L159): The topic entered by the user and the default topic (`/world/worldName/model/modelName/joint_state`). If the user entered topic is invalid, `this->topic` automatically gets set to default one in `this->topic = validTopic(topics)`.

In the second case, `this->topic` will be empty initially and won't be added to the `topics` vector according to [this](https://github.com/jasmeet0915/ign-gazebo/blob/5105afb2f4275987239c2455e964db4e336b46dd/src/systems/joint_state_publisher/JointStatePublisher.cc#L157) check. Therefore, the `topics` vector will only have the default topic which will be returned as it is from `validTopic()`.

The only condition where the suggested check might be required will be if the default topic is invalid which I don't think will be possible (is it possible?).

Reply:
> The only condition where the suggested check might be required will be if the default topic is invalid which I don't think will be possible (is it possible?).

Yup, correct. I think it may be possible that the model name contains characters that can't be handled by the function that generates valid topics.

Reply:
>Yup, correct. I think it may be possible that the model name contains characters that can't be handled by the function that generates valid topics.

Oh yeah that makes sense. 
I have made the changes in the [latest commit](https://github.com/ignitionrobotics/ign-gazebo/pull/1076/commits/281f1c3ecd57e54bc985acc781651fea5da98206)!